### PR TITLE
do not distclean-relx-rel on rel and relx tests

### PR DESCRIPTION
**Pull request by ingwinlu**

### CODE REVIEW

File: test/plugin_relx.mk [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	$t test -d $(APP)/_rel/$(APP)_release/bin
+	$t test -d $(APP)/_rel/$(APP)_release/lib
+	$t test -d $(APP)/_rel/$(APP)_release/releases
+	$t test -d $(APP)/_rel/$(APP)_release/releases/1
+
+relx-run: clean-relx-run
+	$t $(call bootstrap_helper,$(APP))
+
+	$i "Build and create release via relx"
+	$t $(MAKE) -C $(APP)
```
Comment (ethrbh) [inline test/plugin_relx.mk]: hello ingwinlu,

I have started to discuss my problem with Essen, but he has suggested to turn to you instead. So I have an Erlang/OTP project based on erlang.mk. The application up and running, and now I would like to upgrade that with 
_rel/bin/${APP} upgrade

It says a package needs for this without tar.gz suffix. First I started to find how this package is made. Essen has showed your PR where I can see the steps (https://github.com/ninenines/erlang.mk/pull/371/files), but when I tried to make the similar by hand, “nothing” is happened. Please have a look at what I did, and if you have time, please try help me.

	ethrbh@ethrbh-VirtualBox:/mnt/ethrbh/private/projects/github/igs(master)> ls
	bin  board  deps  deps.bak  dev  dummy  ebin  erlang.mk  igs.d  include  Makefile  priv  README.md  rel  _rel  relx  relx.config  self-extractable  src  test  tool_configuration.cfg
	ethrbh@ethrbh-VirtualBox:/mnt/ethrbh/private/projects/github/igs(master)> cat relx.config
	{release, {igs, "0.1.6"}, [igs]}.
	{extended_start_script, true}.
	{sys_config, "rel/sys.config"}.
	{vm_args, "rel/vm.args"}.
	{overlay, [
			%% Create database folder.
			{mkdir, "dbase/Mnesia"},
			
			%% Create dev_patches folder. The patch files can be pushed here.
			{mkdir, "dev_patches"},
			
			%% Copy bin/setup.sh into the bin folder in _rel/ folder
			{copy, "bin/igs.sh", "bin/igs.sh"}
		  ]}.
	ethrbh@ethrbh-VirtualBox:/mnt/ethrbh/private/projects/github/igs(master)> make RELX_OPTS="release tar" -C ../igs/
	make: Entering directory `/mnt/ethrbh/private/projects/github/igs'
	0.1.6
	make: Leaving directory `/mnt/ethrbh/private/projects/github/igs'
	ethrbh@ethrbh-VirtualBox:/mnt/ethrbh/private/projects/github/igs(master)>

thanks for your help,
/Robi

Comment (ingwinlu) [inline test/plugin_relx.mk]: Hi Robi,

keep in mind you have to modify erlang.mk to not delete already built
releases when you create the updated ones. You should be able to
cherry-pick 7817d3ee97241bfa3b8f96ee1173ff70956825fd without much
trouble.

After that the workflow from the included testcase should be able to
build upgrades if you pay attention to the workflow.

-winlu

Comment (ethrbh) [inline test/plugin_relx.mk]: hello ingwinlu,

Thanks for your reply. 

Does this mean the upgrade does not work in the official erlang.mk? Or it works, but the package needs to be made somehow?

So I tried this, where igs is my application. What I don't know, how I can make the package for this.

  /home/ethrbh/igs/bin/igs upgrade 
  Missing package argument
  Usage: igs upgrade {package base name}
  NOTE {package base name} MUST NOT include the .tar.gz suffix

thanks for your help,
/Robi

**Comment by essen** [comment]
The change was already made apparently, on the other hand the tests are very interesting so I will work on merging those in the next few days.

**Comment by essen** [comment]
Almost done updating the tests and merging. Update was needed due to changes in the Erlang.mk test suite, plus some tweaks.

Thanks for the great work, the tests really help me out here.

After merging I will do a few tweaks to the relx plugin to just have "make rel" and "make relup" to obtain the necessary files. I don't think generating a tarball when running "make rel" will be a big problem for most people. On the other hand I will probably change its default location.

**Comment by essen** [comment]
Merged, thanks again!

**Pull request closed without merging**